analysis_driver/dataset.py

```diff
@@ -73,6 +74,10 @@ def fail(self, exit_status):
     def abort(self):
         self.most_recent_proc.finish(DATASET_ABORTED)
 
+    def resume(self):
+        self.terminate()
+        self.most_recent_proc.change_status(DATASET_RESUME)
+
```

Comment:
We should change the terminate method to avoid the race condition where the killed process fail after it has been resumed or reseted  
```python
    def terminate(self):
        pid = self.most_recent_proc.get('pid')
        if pid and self._is_running_pid(pid):
            self.info('Terminating pid %s for %s %s', pid, self.type, self.name)
            os.kill(pid, 10)
        while self._is_running_pid(pid):
            time.sleep(1)


    @staticmethod
    def _is_running_pid(pid):        
        #see http://stackoverflow.com/questions/7653178/wait-until-a-certain-process-knowing-the-pid-end
        try:
            os.kill(pid, 0)
        except OSError as err:
            if err.errno == errno.ESRCH:
                return False
        return True
```